x/emissions/module/rewards/rewards_internal.go

```diff
@@ -225,89 +226,190 @@ func GetStakeWeightedLoss(reputersStakes, reputersReportedLosses []alloraMath.De
 	return stakeWeightedLoss, nil
 }
 
-// GetStakeWeightedLossMatrix calculates the stake-weighted
-// geometric mean of the losses to generate the consensus vector.
-// L_i - consensus loss vector
-func GetStakeWeightedLossMatrix(
-	reputersAdjustedStakes []alloraMath.Dec,
-	reputersReportedLosses [][]alloraMath.Dec,
-) ([]alloraMath.Dec, []alloraMath.Dec, error) {
-	if len(reputersAdjustedStakes) == 0 || len(reputersReportedLosses) == 0 {
-		return nil, nil, types.ErrInvalidSliceLength
+// GetStakeWeightedLossMatrix returns the consensus loss and most distant loss for each column
+func GetStakeWeightedLossMatrix(stakes []alloraMath.Dec, losses [][]alloraMath.Dec) ([]alloraMath.Dec, []alloraMath.Dec, error) {
+	if len(stakes) == 0 || len(losses) == 0 {
+		return nil, nil, fmt.Errorf("empty stakes or losses")
+	}
+	if len(stakes) != len(losses) {
+		return nil, nil, fmt.Errorf("stakes and losses must have same length")
 	}
-	var err error
 
-	// Ensure every loss array is non-empty and calculate geometric mean
-	stakeWeightedLoss := make([]alloraMath.Dec, len(reputersReportedLosses[0]))
-	mostDistantValues := make([]alloraMath.Dec, len(reputersReportedLosses[0]))
-	for j := 0; j < len(reputersReportedLosses[0]); j++ {
-		// Calculate total stake to consider
-		// Skip stakes of reputers with NaN losses
-		totalStakeToConsider := alloraMath.ZeroDec()
-		for i, losses := range reputersReportedLosses {
-			// Skip if loss is NaN
-			if losses[j].IsNaN() {
-				continue
+	// Get the number of columns (predictions)
+	numCols := len(losses[0])
+	for _, row := range losses {
+		if len(row) != numCols {
+			return nil, nil, fmt.Errorf("all loss rows must have same length")
+		}
+	}
+
+	// First check if all stakes are zero or epsilon
+	allZeroOrEpsilon := true
+	for _, stake := range stakes {
+		if !stake.IsZero() && !stake.Equal(alloraMath.EpsilonDec()) {
```

Comment:
It'd be better to have a goal var with the epsilon dec instead of allocating a new one each iterations